src/store/sagas/exchange.ts

```diff
@@ -611,21 +617,204 @@ export function* handleSwaplineSwap(): Generator {
   }
 }
 
+export function* handleAddCollateralVault(vaultSwapData: VaultSwap): SagaGenerator<string> {
+  const wallet = yield* call(getWallet)
+  const exchangeProgram = yield* call(getExchangeProgram)
+  const tokensAccounts = yield* select(accounts)
+  const userCollateralTokenAccount = tokensAccounts[vaultSwapData.collateral.toString()]
+  const vaultsPair = yield* select(vaults)
+  const connection = yield* call(getConnection)
+  const token = new Token(
+    connection,
+    tokensAccounts[vaultSwapData.collateral.toString()].address,
+    TOKEN_PROGRAM_ID,
+    new Account()
+  )
+  const userVaultState = yield* select(userVaults)
+  const { ix } = yield* call([exchangeProgram, exchangeProgram.createVaultEntryInstruction], {
+    owner: wallet.publicKey,
+    synthetic: vaultSwapData.synthetic,
+    collateral: vaultSwapData.collateral
+  })
+  const depositTx = yield* call([exchangeProgram, exchangeProgram.vaultDepositTransaction], {
+    collateral: vaultSwapData.collateral,
+    synthetic: vaultSwapData.synthetic,
+    owner: wallet.publicKey,
+    amount: vaultSwapData.collateralAmount,
+    userCollateralAccount: userCollateralTokenAccount.address,
+    reserveAddress: vaultsPair[vaultSwapData.vaultAddress.toString()].collateralReserve,
+    collateralToken: token
+  })
+  let tx
+
+  if (!vaultSwapData.vaultEntryExist) {
+    tx = new Transaction().add(ix).add(depositTx)
+  } else {
+    tx = new Transaction().add(depositTx)
+  }
+
+  const signature = yield* call(signAndSend, wallet, tx)
+  yield* call(sleep, 1500) // Give time to subscribe to account
+  if (typeof userVaultState[vaultSwapData.vaultAddress.toString()] === 'undefined') {
+    yield* put(
+      actionsVault.setNewVaultEntryAddress({
+        newVaultEntryAddress: vaultSwapData.vaultAddress
+      })
+    )
+  }
```

Comment:
you first need to add this to redux then give time for subscription.
You done this other way around.